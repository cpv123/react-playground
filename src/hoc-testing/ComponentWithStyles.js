import React from 'react';
import { 
    Button,
    ExpansionPanel,
    ExpansionPanelSummary,
    ExpansionPanelDetails,
    Typography,
    withStyles
} from '@material-ui/core';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';

const styles = theme => ({
    root: {
        width: '100%',
    },
    heading: {
        fontSize: theme.typography.pxToRem(15),
        fontWeight: theme.typography.fontWeightRegular,
    },
    button: {
        margin: theme.spacing.unit, 
    },
});

export function Component({ classes, children }) {
    return (
        <div className={classes.root}>
            <ExpansionPanel>

                <ExpansionPanelSummary expandIcon={<ExpandMoreIcon />}>
                    <Typography className={classes.heading}>
                        Expansion Panel Title
                    </Typography>
                </ExpansionPanelSummary>

                <ExpansionPanelDetails>
                    {children}
                    <Button 
                        className={classes.button}
                        color="primary"
                        variant="contained"
                        onClick={() => console.log('You clicked a well styled button')}
                    >
                        Click Me
                    </Button>
                </ExpansionPanelDetails>

            </ExpansionPanel>
        </div>
    );
}

const ComponentWithStyles = withStyles(styles)(Component);

export default ComponentWithStyles;
