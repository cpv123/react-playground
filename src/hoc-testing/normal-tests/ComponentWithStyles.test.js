import React from 'react';
import { shallow } from 'enzyme';
import ComponentWithStyles from '../ComponentWithStyles';

describe('The ComponentWithStyles component', () => {

    it('renders as expected', () => {
        const wrapper = shallow(<ComponentWithStyles />);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders children passed in', () => {
        const wrapper = shallow(
            <ComponentWithStyles>
                <span>Click the button below!</span>
            </ComponentWithStyles>
        );
        expect(wrapper.find('span')).toHaveLength(1);
    })

});