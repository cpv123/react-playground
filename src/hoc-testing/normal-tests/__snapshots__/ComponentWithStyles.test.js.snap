// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ComponentWithStyles component renders as expected 1`] = `
<Component
  classes={
    Object {
      "button": "Component-button-3",
      "heading": "Component-heading-2",
      "root": "Component-root-1",
    }
  }
/>
`;
